crates/common/src/zksolc_manager.rs

```diff
@@ -674,3 +725,77 @@ impl ZkSolcManager {
         Ok(())
     }
 }
+
+/// Get minimal requirements of the compiler version for the zksolc compiler.
+///
+/// This function downloads the JSON file from the zksolc-bin repository and
+/// returns the minimal requirements of the compiler version for the zksolc
+/// compiler.
+///
+/// # Returns
+///
+/// [`SolcMetadata`]
+///
+/// # Errors
+///
+/// The error may occure on failed connection establishing to repository
+/// provider. Also, if path or structure of `version.json` file changes,
+/// the error may occure too.
+pub async fn fetch_solc_metadata() -> Result<SolcMetadata> {
+    let client = Client::new();
+    let response = client
+        .get(ZKSOLC_VERSIONS_DOWNLOAD_URL)
+        .send()
+        .await
+        .map_err(|e| Error::msg(format!("Failed to get solc versions file: {e}")))?;
+
+    if !response.status().is_success() {
+        return Err(Error::msg(format!(
+            "Failed to get solc versions file: status code {}",
+            response.status()
+        )))
+    }
+
+    let solc_metadata: SolcMetadataResponseBody = response
+        .json()
+        .await
+        .map_err(|e| Error::msg(format!("Failed to parse solc versions file: {e}")))?;
+
+    solc_metadata.try_into()
+}
+
+/// Metadata of the solc-bin comiler.
+// TODO(Velnbur): handle invalid semver spec structure by using types from
+// `semver` crate or implement proper deserialize for `ZkSolcVersion`.
+#[derive(serde::Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SolcMetadataResponseBody {
+    /// The latest supported version of solc compiler as a string in semver spec.
+    pub(crate) latest: String,
+
+    /// The minimal supported version of solc compiler as a string in semver spec.
+    pub(crate) min_version: String,
+}
+
+/// Metadata of the solc-bin comiler.
+#[derive(Clone, Debug)]
+pub struct SolcMetadata {
+    /// The latest supported version of solc compiler
+    pub latest: ZkSolcVersion,
+
+    /// The minimal supported version of solc compiler
+    pub min_version: ZkSolcVersion,
+}
+
+impl TryFrom<SolcMetadataResponseBody> for SolcMetadata {
+    type Error = anyhow::Error;
+
+    fn try_from(value: SolcMetadataResponseBody) -> Result<Self, Self::Error> {
+        Ok(SolcMetadata {
+            // HACK: the `version.json` file has version without `v` prefix, but
+            // `parse_version` requires them, we so we add them this `format!`.
```

Comment:
@Velnbur can you please update comment to indicate a `TODO` please. Otherwise this LGTM!

Reply:
done